lasagne/objectives.py

```diff
@@ -341,45 +342,50 @@ def multiclass_hinge_loss(predictions, targets, delta=1):
 def huber_loss(predictions, targets, delta=1):
     """ Computes the huber loss between predictions and targets.
 
-    .. math:: L_\delta (diff) = \frac{diff^2}{2} & \text{if  } |diff| \le \
-                                \delta
-    .. math:: L_\delta (diff) & = & \delta (|diff| - \frac{\delta}{2} ),\
+    .. math:: 
+        L_\delta (diff) = \frac{diff^2}{2} & \text{if  } |diff| \le \ delta \\
+
+        L_\delta (diff) & = & \delta (|diff| - \frac{\delta}{2} ),\ \\
                                      &\text{else}
 
     Parameters
     ----------
-    predictions : Theano 2D tensor
-        Predictions in (0, 1), such as softmax output of a neural network,
-        with data points in rows and class probabilities in columns.
+    predictions : Theano 2D tensor or 1D tensor
+        Predictions output of a neural network
+
     targets : Theano 2D tensor or 1D tensor
-        Either a vector of int giving the correct class index per data point
-        or a 2D tensor of one-hot encoding of the correct class in the same
-        layout as predictions (non-binary targets in [0, 1] do not work!)
+        Ground truth to which the prediction is to be compared
+        with.
```

Comment:
Yes, or you can also call it `a` and `b`, since the loss is completely symmetric. Your current docstring says they should be 1D or 2D tensors, which is too strict! They can be anything -- it's a drop-in replacement for the squared error.